File: frontend/src/services/api/outline.ts
import apiClient from './client';

export interface OutlineSection {
  level: number;
  title: string;
  content: string;
  order: number;
  subsections?: OutlineSection[];
}

export interface Outline {
  id: number;
  user_id: number;
  project_id: number;
  title: string;
  content: {
    title: string;
    sections: OutlineSection[];
  };
  version: number;
  created_at: string;
  updated_at: string;
}

export interface GenerateOutlineRequest {
  project_id: number;
  topic_title: string;
  requirements?: string;
}

export interface UpdateOutlineRequest {
  title?: string;
  content?: {
    title: string;
    sections: OutlineSection[];
  };
}

export interface OutlineListResponse {
  items: Outline[];
  total: number;
  page: number;
  page_size: number;
}

export const outlineApi = {
  generate: (data: GenerateOutlineRequest) =>
    apiClient.post<Outline>('/outlines/generate', data),

  list: (params: { project_id?: number; page?: number; page_size?: number }) =>
    apiClient.get<OutlineListResponse>('/outlines', { params }),

  get: (id: number) => apiClient.get<Outline>(`/outlines/${id}`),

  update: (id: number, data: UpdateOutlineRequest) =>
    apiClient.put<Outline>(`/outlines/${id}`, data),

  delete: (id: number) => apiClient.delete(`/outlines/${id}`),
};

File: backend/app/schemas/outline.py
"""
大纲Schemas
"""
from datetime import datetime
from typing import Optional, Any, List
from pydantic import BaseModel, Field


class OutlineBase(BaseModel):
    """大纲基础Schema"""
    title: str = Field(..., max_length=500)
    content: Optional[Any] = None  # JSON 格式的大纲内容


class OutlineCreate(OutlineBase):
    """大纲创建Schema"""
    project_id: int


class OutlineUpdate(BaseModel):
    """大纲更新Schema"""
    title: Optional[str] = Field(None, max_length=500)
    content: Optional[Any] = None


class OutlineResponse(OutlineBase):
    """大纲响应Schema"""
    id: int
    user_id: int
    project_id: int
    version: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class OutlineGenerateRequest(BaseModel):
    """大纲生成请求Schema"""
    project_id: int = Field(..., description="项目ID")
    topic_title: str = Field(..., description="选题标题")
    requirements: Optional[str] = Field(None, description="额外要求")


class OutlineListResponse(BaseModel):
    """大纲列表响应Schema"""
    items: List[OutlineResponse]
    total: int
    page: int
    page_size: int
